botorch/test_functions/base.py

```diff
@@ -180,8 +185,8 @@ def __init__(
                 f"If specified as a list, length of noise_std ({len(noise_std)}) "
                 f"must match the number of objectives ({len(self._ref_point)})"
             )
-        super().__init__(noise_std=noise_std, negate=negate)
-        ref_point = torch.tensor(self._ref_point, dtype=torch.get_default_dtype())
+        super().__init__(noise_std=noise_std, negate=negate, dtype=dtype)
+        ref_point = torch.tensor(self._ref_point, dtype=self.dtype)
```

Comment:
Note: I spotted another use of `torch.get_default_dtype` while implementing my changes - I guess that this should be changed, only question is to what. For now, I used `self.dtype`, but I guess it should rather be `X.dtype`?

Reply:
At this point during init we don't know what `X` will be - so I think the only reasonable thing to do here would be to be consistent with how we handle the `bounds`.

This raises the question what to do when getting an `X` with a different `dtype` during `forward` or `evaluate_true`. But I think we should be able to just rely on type promotion there. 


```suggestion
        ref_point = torch.tensor(self._ref_point, dtype=dtype)
```